Geant4star integration

klendathu2k opened the pull request:
Following suggestions made in PR#311...  splitting the geant4star pull request.  First committing the changes to BigFullChain.h that define the chain options needed to load in geant4 and geant4vmc support libraries.  Also adding the environment setup script and build script under StGeant4Maker/etc.  And a test macro to load the libraries.

To compile:
$ source StRoot/StGeant4Maker/etc/setup
$ source StRoot/StGeant4Maker/etc/build

To test:
$ root.exe StRoot/StGeant4Maker/macros/loadGeant4Vmc.C

Result should be a successful load of the libraries.

klendathu2k committed:
This PR fixes #324.

Before submitting PR#315 some code cleanup was performed (see note below).
The fast jet directory was moved from  StarGenerator/FastJetFilt   to
StarGenerator/FILT/FastJetFilt, to conform with the standard source tree
layout for filters.

https://github.com/klendathu2k/star-sw-1/commit/4f270342c76226ae6e33e732cad06fad37d1b10e
https://github.com/klendathu2k/star-sw-1/commit/625bbddaa2990946c57a9d6014c164378d7c4b32

Conscript-standard directory rules allow one to specify include paths
based on the module (StarGenerator) and package.  The package is one
directory beneath the module...  before code cleanup this was the
FastJetFilter directory.  After cleanup, we have to apply the fast jet
include path to all filters in the StarGenerator/FILT directory.

----

Note on the pp 200 HF filtered jet production.  The production commenced with
fast jet filter code built before code cleanup, based on the original
location of the fast jet filter in the source tree.

klendathu2k committed:
Remove debug statement

klendathu2k committed:
Merge branch 'star-bnl:main' into main

klendathu2k committed:
Merge branch 'star-bnl:main' into main

klendathu2k committed:
Merge branch 'star-bnl:main' into main

klendathu2k committed:
Merge branch 'star-bnl:main' into main

klendathu2k committed:
Merge branch 'star-bnl:main' into main

klendathu2k committed:
Merge branch 'star-bnl:main' into main

klendathu2k committed:
Merge branch 'star-bnl:main' into main

klendathu2k committed:
Environment setup and build script for geant4star

klendathu2k committed:
Support for geant 4, geant4 vmc, geant3 vmc, geant4maker, ...

klendathu2k committed:
Test macro to load geant4 VMC libraries

klendathu2k committed:
Reduce number of chain options to load G4 libraries

klendathu2k commented on StRoot/StBFChain/BigFullChain.h:
```diff
@@ -1211,6 +1211,7 @@ Bfc_st BFC[] = { // standard chains
   {"McEvOut"     ,""  ,"","StMcEvent,Tree"                       ,"","","Write StMcEvent to StTree",kFALSE},
   {"EvOut"       ,""  ,"","Tree"                                   ,"","","Write StEvent to StTree",kFALSE},
   {"GeantOut"    ,""  ,"","Tree"                                ,"","","Write g2t tables to StTree",kFALSE},
+  {"Geant4Out"   ,""  ,"","Tree"                        ,"","","Write g2t tables to StTree from G4",kFALSE},
```
Note: this chain option requires an update to StBFChain.cxx to add the StGeant4Maker into the tree, which must
wait for the integration of StGeant4Maker.

klendathu2k commented on StRoot/StBFChain/BigFullChain.h:
```diff
@@ -1377,12 +1378,38 @@ Bfc_st BFC[] = { // standard chains
+  {"stargen.util", "","", "gen_T,sim_T"/*+++*/,                     "", "libVMC.so,libStarGeneratorUtil.so,libMathMore.so","STAR Generator BASE",false},
+  {"stargen.event","","", "gen_T,sim_T"/*+++*/,                     "", "libVMC.so,libStarGeneratorEvent.so,libMathMore.so","STAR Generator BASE",false},
+  {"stargen.base", "","", "gen_T,sim_T"/*+++*/,                     "", "libVMC.so,libStarGeneratorBase.so,libMathMore.so","STAR Generator BASE",false},
+  {"stargen",     "","",  "stargen.util,stargen.event,stargen.base"/*+++*/,                     "", "","STAR Generator BASE",false},
```
Note: stargen loads have an unmet dependency in St_geant_Maker.  Future commits will resolve this.

starsdong commented:
Jason and all, I see this one has been there for a while. As far as I see, all checks have passed, so the merge won't crash the running jobs? From your comments, Jason, there will still need other commits to resolve some of the dependencies. How should we proceed? Shall we just proceed to merge this PR (upon the sync with the latest main branch)? Thanks

klendathu2k committed:
Merge branch 'main' into geant4star-integration

klendathu2k committed:
Merge branch 'main' into geant4star-integration

klendathu2k committed:
Update to clhep 2.4.5.1

klendathu2k committed:
Build stic first... build root4star at end for testing

klendathu2k committed:
Added a README.md file... mainly so that I can document the push
that is about to happen...

Updated the setup and build scripts for the current pull request.
The clhep version changed out from under us since the PR was made.
Also, the build script now constructs root4star and starsim for
general testing.

klendathu2k committed:
Merge branch 'geant4star-integration' of https://github.com/klendathu2k/star-sw-1 into geant4star-integration

plexoos commented on StRoot/StGeant4Maker/etc/setup:
```diff
@@ -0,0 +1,25 @@
+setup 64b
+
+module use /cvmfs/star.sdcc.bnl.gov/star-spack/spack/share/spack/modules/linux-rhel7-x86_64
+#module load root-6.24.00
+source /star/simu/simu/jwebb/ROOT/local/bin/thisroot.csh
```
Can you use the the ROOT installed by Spack instead? This should suffice:

```
module load star-env-X.Y.Z-root-6.V.W
```

...and you don't need to set the environment variables by hand as you do later in this file.

klendathu2k commented:
As I recall, I encountered problems when trying to compile against that
environment.  Probably a missing ROOT feature... and/or needed to have a
more modern VMC version than was compiled.  These are issues I would 
prefer
to address in a future PR.

On 2022-09-29 12:26, Dmitri Smirnov wrote:
> @plexoos commented on this pull request.
> -------------------------
> 
> In StRoot/StGeant4Maker/etc/setup [1]:
> 
>> +#module load root-6.24.00
> +source /star/simu/simu/jwebb/ROOT/local/bin/thisroot.csh
> 
> Can you use the the ROOT installed by Spack instead? This should
> suffice:
> 
> module load star-env-X.Y.Z-root-6.V.W
> 
> ...and you don't need to set the environment variables by hand as you
> do later in this file.
> 
> --
> Reply to this email directly, view it on GitHub [2], or unsubscribe
> [3].
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> 
> Links:
> ------
> [1] https://github.com/star-bnl/star-sw/pull/377#discussion_r983758098
> [2] 
> https://github.com/star-bnl/star-sw/pull/377#pullrequestreview-1125641979
> [3] 
> https://github.com/notifications/unsubscribe-auth/ANL4LVEHDMY7OI5EP7AX4ATWAW7KPANCNFSM53M2TREA

klendathu2k commented:
Superseded by #617
